src/cpp/rtps/reader/StatefulReader.cpp

```diff
@@ -587,14 +587,20 @@ bool StatefulReader::processGapMsg(
         // TODO (Miguel C): Refactor this inside WriterProxy
         SequenceNumber_t auxSN;
         SequenceNumber_t finalSN = gapList.base() - 1;
+        History::const_iterator history_iterator = mp_history->changesBegin();
         for (auxSN = gapStart; auxSN <= finalSN; auxSN++)
         {
             if (pWP->irrelevant_change_set(auxSN))
             {
-                CacheChange_t* to_remove = findCacheInFragmentedProcess(auxSN, pWP->guid());
+                CacheChange_t* to_remove = nullptr;
+                auto ret_iterator = findCacheInFragmentedProcess(auxSN, pWP->guid(), &to_remove, history_iterator);
                 if (to_remove != nullptr)
                 {
-                    mp_history->remove_change(to_remove);
+                    ret_iterator = mp_history->remove_change_nts(to_remove, ret_iterator);
```

Comment:
Perhaps it should be `history_iterator = mp_history->remove_change_nts(to_remove, ret_iterator);`

Reply:
You're right, done